framework/test/mocha/unit/modules/chain/blocks/blocks.js

```diff
@@ -888,12 +922,85 @@ describe('blocks', () => {
 		});
 
 		it('should update the broadhash', async () => {
-			await blocksInstance._processReceivedBlock(block);
+			await blocksInstance._processReceivedBlock[block.version](block);
 			expect(stubs.updateBroadhash).to.be.called;
 		});
 
 		it('should assign the new last block to _lastBlock and _isActive to false', async () => {
-			await blocksInstance._processReceivedBlock(block);
+			await blocksInstance._processReceivedBlock[block.version](block);
+			expect(blocksInstance._lastBlock).to.equal(block);
+			expect(blocksInstance._isActive).to.be.false;
+		});
+
+		it('should error log that apply the new received block failed with the error', async () => {
+			const error = new Error('This is an error');
```

Comment:
Please use // Arrange & Act & Assert comments when test gets bigger. 

You can also move to error message in line 944 to arrange part as a variable, to make assertion easy to read.

Reply:
Done